Lib/glyphsLib/builder.py

```diff
@@ -371,6 +374,49 @@ def set_custom_params(ufo, parsed=None, data=None, misc_keys=(), non_info=()):
             ufo.lib[GLYPHS_PREFIX + name] = value
 
 
+def parse_glyphs_filter(filter_str):
+    """Parses glyphs custom filter string into a dict object that
+       ufo2ft can consume.
+
+        Reference:
+            ufo2ft: https://github.com/googlei18n/ufo2ft
+            Glyphs 2.3 Handbook July 2016, p184
+
+        Args:
+            filter_str - a string of glyphs app filter
+
+        Return:
+            A dictionary contains the structured filter.
+            Return None if parse failed.
+    """
+    elements = filter_str.split(';')
+
+    if len(elements) == 0 or elements[0] == '':
```

Comment:
if `filter_str ` is a string (even an empty one), and you call `.split()` method on it, the resulting `len(elements)` can never be 0; it'll always be at least 1 (the list would contain the original string, if the separator is not present; or more than one substring).